### PR TITLE
String util

### PR DESCRIPTION
**Pull request by ulmus-scott**
Originally part of https://github.com/MythTV/mythtv/pull/438
depends on https://github.com/MythTV/mythtv/pull/439

##### Checklist

- [x] contribution is in a branch rebased against [master](https://github.com/MythTV/mythtv)
- [x] code compiles successfully without errors
- [x] code follows the [MythTV Coding Standards](https://www.mythtv.org/wiki/Coding_Standards)
- [x] commits are logically organised and have [good commit messages](https://chris.beams.io/posts/git-commit)

### CODE REVIEW

**Comment by ulmus-scott** [comment]
Rebased and added https://github.com/MythTV/mythtv/pull/440/commits/853ce611db4dabb0e2159b00fade10d110a78f4d

**Comment by ulmus-scott** [comment]
Rebased onto master.

Fixup create mythrandom is now in https://github.com/MythTV/mythtv/pull/460

File: mythtv/libs/libmythbase/stringutil.cpp [hunk 1]
```diff
@@ -1,30 +1,134 @@
...
 
-#if QT_VERSION < QT_VERSION_CHECK(6,0,0)
-#include <QTextCodec>
+#if __has_include(<bit>) // C++20
+#include <bit>
+#endif
+
+#include <climits> // for CHAR_BIT
+
+#if __cpp_lib_bitops >= 201907L
```
Comment (ulmus-scott) [inline mythtv/libs/libmythbase/stringutil.cpp:9]: This raises a -Wundef.  I will silence it once the GitHub Action is completed.

**Comment by ulmus-scott** [comment]
Rebased onto master.

**Comment by linuxdude42** [comment]
Please resolve the conflicts and remove the reordering of includes. Other than that it looks good. Sorry it took so long to review.

**Pull request merged**